microsoft/testsuites/cvm/cvm_boot.py

```diff
@@ -0,0 +1,176 @@
+        )
+        if not security_profile_settings.encrypt_disk:
+            raise SkippedException("This test requires disk encryption to be enabled")
+        lsblk = node.tools[Lsblk]
+        disks = lsblk.get_disks(force_run=True)
+        partitions: List[PartitionInfo] = next(
+            (d.partitions for d in disks if d.name == "sda"), []
+        )
+        assert_that(partitions, "Cannot find a disk named 'sda'").is_not_empty()
+        root_partition = next((p for p in partitions if p.name == "sda2"), None)
+        assert_that(root_partition, "Cannot locate root partition").is_not_none()
+        assert isinstance(root_partition, PartitionInfo)
+        assert_that(root_partition.fstype).is_equal_to("crypto_LUKS")
+
+    @TestCaseMetadata(
+        description="""This test case verifies that a CVM can still boot if any boot
+        component is upgraded.
+
+        Steps:
+        1. On first boot, check current PCR values for PCR4 and PCR7
+        2. Get current boot components versions (e.g. shim, grub, systemd-boot, uki)
+        3. Run a package upgrade to update boot components
+        4. Get new boot components versions to see if anything has changed
+        5. Reboot the CVM, make sure the CVM can boot up again
+        6. PCR4 should change if any of the boot components is upgraded
+        7. PCR7 may change (for example, if a signing certificate is changed)
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[CvmEnabled()],
+            supported_platform_type=[AZURE],
+        ),
+    )
+    def verify_boot_success_after_component_upgrade(
+        self,
+        log: Logger,
+        node: RemoteNode,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+        **kwargs: Any,
+    ) -> None:
+        posix_os: Posix = cast(Posix, node.os)
+        # First boot
+        # - Check PCR values (PCR4, PCR7)
+        pcrs_before_reboot = node.tools[Tpm2].pcrread(pcrs=[4, 7])
+
+        # - Get current boot components versions (shim, systemd-boot, kernel-uki)
+        boot_components = ["shim", "systemd-boot", "kernel-uki"]
+        boot_components_versions = dict()
+        for pkg in boot_components:
+            pkg_info = posix_os.query_package(pkg)
+            boot_components_versions[pkg] = pkg_info.version_str
+
+        # - Upgrade boot components
+        posix_os.update_packages(boot_components)
+
+        # - Get new boot components versions
+        boot_components_new_versions = dict()
+        for pkg in boot_components:
+            pkg_info = posix_os.query_package(pkg)
+            boot_components_new_versions[pkg] = pkg_info.version_str
+
+        # Reboot
+        # - Make sure VM boots up again
+        reboot_tool = node.tools[Reboot]
+        reboot_tool.reboot_and_check_panic(log_path)
+        is_ready, tcp_error_code = wait_tcp_port_ready(
```

Comment:
It doesn't need to wait for ready in the test case. When you call next command, LISA will do it.

Reply:
Do you mean the `wait_tcp_port_ready` call can be removed? I was mostly following the pattern in smoke_test and a boot test defined in microsoft/testsuites/core/boot.py

Reply:
The test logic of smoke_test is different with most cases, because it doesn't need SSH port 22 working. The cvm test case must have the port 22, so it can follow other cases on handling reboot.